marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/steps/StepController.java

```diff
@@ -0,0 +1,46 @@
+package com.marklogic.hub.central.controllers.steps;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.marklogic.hub.central.controllers.BaseController;
+import com.marklogic.hub.dataservices.ArtifactService;
+import io.swagger.annotations.ApiOperation;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+@Controller
+@RequestMapping("/api/steps")
+public class StepController extends BaseController {
+    @RequestMapping(method = RequestMethod.GET)
+    @ApiOperation(value = "Get all steps grouped by type", response = Steps.class)
+    @Secured("ROLE_readFlow")
+    public ResponseEntity<JsonNode> get() {
+        ObjectNode response = new ObjectMapper().createObjectNode();
+        ArtifactService service = newArtifactService();
+        response.set("ingestionSteps", service.getList("ingestion", Stream.of("stepId", "name"), false));
```

Comment:
Pinging @bsrikan and @srinathgit  about this - the issue with our use of "name" on a step document will impact an entity "name" property with a range index on it (flow documents do the same). 

I am thinking we could kill two birds with one stone here - I think it's very common to need the ID and name from a set of documents, and thus it's common to add range indexes to those. "stepId" is a good name, but "name" of course is not for a range index. 

How about if - as part of migration for 5.3 - we use "stepName" instead of "name" in step documents and add range indexes for stepId and stepName to the staging and final databases? I think that's a good use of indexes. We'd then just do a new DS endpoint that retrieves the ID and name properties for all steps (it looks like "false" is always the 3rd parameter, so that's not really needed?).

Reply:
@rjrudin ,
I am ok with changing "name" to "stepName" and having range index on "stepName" and "stepId". In addition to flow migration, we will also have to change it back to "name" in getFullFlow() method  (in core.sjs)

Reply:
Is this something to cover in this PR?

Reply:
I think it would be a separate JIRA ticket so we could fully think through it. It does seem worthwhile to change "name" to "stepName" as part of 5.3.0.